core/src/main/java/org/elasticsearch/action/search/type/TransportSearchTypeAction.java

```diff
@@ -115,7 +115,10 @@ protected BaseAsyncAction(SearchRequest request, ActionListener<SearchResponse>
 
             clusterState.blocks().globalBlockedRaiseException(ClusterBlockLevel.READ);
 
-            String[] concreteIndices = indexNameExpressionResolver.concreteIndices(clusterState, request.indicesOptions(), request.indices());
+            // TODO: I think startTime() should become part of ActionRequest and that should be used both for index name
+            // data math expressions and $now in scripts. This way all apis will deal with now in the same way instead
+            // of just for the _search api
+            String[] concreteIndices = indexNameExpressionResolver.concreteIndices(clusterState, request.indicesOptions(), startTime(), request.indices());
```

Comment:
this //TODO will have to stay right? maybe we want to open an issue instead to keep track of this? also s/data/date